env0/resource_environment.go

```diff
@@ -87,7 +87,7 @@ func resourceEnvironment() *schema.Resource {
 			},
 			"auto_deploy_by_custom_glob": {
 				Type:        schema.TypeString,
-				Description: "redeploy on file filter pattern",
+				Description: "redeploy on file filter pattern. If used then 'auto_deploy_on_path_changes_only' must be configured to true. If used then 'deploy_on_push' or 'run_plan_on_pull_requests' must be configured to true.",
```

Comment:
Some suggestions that might improve the description. feel free to ignore this comment:
- Adding `'\n'` like you did in `templateId`
- Changing "If used then" to "When used"